### PR TITLE
Fix None value crash for orientation update

### PR DESCRIPTION
**Pull request by potsui**
@stvstnfrd @caesar2164 @caseylitton PR for fixing `TypeError: object of type 'NoneType' has no len()` on crash list

Previously, when a user uploads a profile picture with EXIF data without an `orientation` value (embedded image data that specifies how an image is oriented), an uncaught `TypeError: object of type 'NoneType' has no len()` would crash the server.

This fix ensures `orientation` is only updated when its value exists in the original image EXIF, and adds a test to check an image without `orientation` will not throw an exception.

Trello: https://trello.com/c/nYQmFFEj/598-crash-list-typeerror-object-of-type-nonetype-has-no-len
Cherry-picked from upstream: https://github.com/edx/edx-platform/commit/178e497c65ec2f81db9ca8ccd6a1f83d7cd3fd3b

CC: @jspayd @jlikhuva

### CODE REVIEW

File: openedx/core/djangoapps/profile_images/tests/test_images.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
     validate_uploaded_image,
     _get_exif_orientation,
     _get_valid_file_types,
+    _update_exif_orientation,
```
Comment (stvstnfrd) [inline openedx/core/djangoapps/profile_images/tests/test_images.py]: Heh, this is technically correct, but this is one of the few times where we don't want to make this kind of change. In this case, this is a line added by edX. While we would prefer they always include the comma, sadly, they don't. Even though including it is the "right thing" to do, we want to avoid diverging from edX. In this case, us adding this when edX doesn't, means there's an increased chance that we end up with a merge conflict on this line, the next time we pull code from edx.
As such, let's revert this change.

File: openedx/core/djangoapps/profile_images/images.py [hunk 2]
```diff
@@ -205,7 +205,8 @@ def _update_exif_orientation(exif, orientation):
     the exif orientation, return a new exif with the orientation set.
     """
     exif_dict = piexif.load(exif)
-    exif_dict['0th'][piexif.ImageIFD.Orientation] = orientation
+    if orientation:
+        exif_dict['0th'][piexif.ImageIFD.Orientation] = orientation
```
Comment (caesar2164) [inline openedx/core/djangoapps/profile_images/images.py]: apparently EXIF orientations are between 1-8 or 0 if none is supplied [1]. Could we autodefault this to 0 if none is supplied?

```python
exif_dict['0th'][piexif.ImageIFD.Orientation] = orientation or 0
```

[1] https://docs.imgix.com/apis/url/rotation/or

Comment (caesar2164) [inline openedx/core/djangoapps/profile_images/images.py]: obviously the tests should be updated to check for this...

Comment (stvstnfrd) [inline openedx/core/djangoapps/profile_images/images.py]: @caesar2164 This is a cherry-pick from upstream.

Comment (caesar2164) [inline openedx/core/djangoapps/profile_images/images.py]: nevermind

**Comment by potsui** [comment]
P.S. To test, upload this as your profile pic: https://drive.google.com/open?id=0BySlmLai5rG1bEhfd0JfbzNLT1U. There should be no error message.

For funsies you can also test with your own images by setting the EXIF orientation value to `None`. You can do that here: https://www.thexifer.net/#exif-camera-settings
